Framework/API/src/NearestNeighbours.cpp

```diff
@@ -141,18 +126,16 @@ void NearestNeighbours::build(const int noNeighbours) {
   BoundingBox bbox;
   // Base the scaling on the first detector, should be adequate but we can look
   // at this
-  IDetector_const_sptr firstDet = (*spectraDets.begin()).second;
-  firstDet->getBoundingBox(bbox);
+  const auto &firstDet = m_spectrumInfo.detector(indices.front());
```

Comment:
Also, I do not really like such a concept because it is really dangerous. There is no way of defining such a detector in an arbitrary case (apart from the fact that we actually want to remove as much direct access as possible).